mpyg321/mpyg321.py

```diff
@@ -1,6 +1,28 @@
 import pexpect
 from threading import Thread
 
+def check_installed():
+    """Checks if mpg321 or mpg123 is installed"""
+    import os
+
+    path = os.environ["PATH"].split(os.pathsep)
+    for directory in path:
+        mpg321_path = os.path.join(directory, "mpg321")
```

Comment:
It's a nice idea to use os.path. I was actually thinking of another way to do this with exception handling. Basically, you can try to run mpg321 from the init function. If it works, you continue executing. If there is an exception, then you try mpg123. And if both fails, you show the error message.

Here, you are kind of rewriting the OS logic for loading a binary, which is a bit overkill don't you think?

Reply:
I guess it is a bit overkill. My thinking was that I didn't want to run the actual program since I then the output from the program would be printed every time the user would try to run it. I didn't know you could pipe the output to a null device using subprocess. I've removed this function and replaced it with a new function that does what you suggested, which is probably better.